### PR TITLE
setup: add mailchimp deps

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: backend/inspirehep/mailing/cli.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import structlog
+from flask.cli import with_appcontext
+
+from .api.jobs import (
+    get_jobs_from_last_week,
+    get_jobs_weekly_html_content,
+    send_jobs_weekly_campaign,
+)
+
+LOGGER = structlog.getLogger()
```
Comment (pazembrz) [inline backend/inspirehep/mailing/cli.py]: Are you using this logger?

File: backend/inspirehep/mailing/cli.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            click.secho("No jobs found from last week skipping...", fg="red")
+            return
+        click.secho(f"Found {len(jobs)} job records from last week.", fg="green")
+
+        content = get_jobs_weekly_html_content(jobs)
+        click.secho("Sending the campaign...", fg="green")
+
+        send_jobs_weekly_campaign(content, test_emails=emails)
+    except Exception as e:
+        click.secho(f"Failed {e}", fg="red")
```
Comment (pazembrz) [inline backend/inspirehep/mailing/cli.py]: Don't you also want to log it to sentry?

File: backend/inspirehep/mailing/providers/mailchimp.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    Retruns:
+        str: the new campaign id.
+    Note:
+        The campaigns can be retrieved by running:
+        > MailChimp(mc_api=MAILCHIMP_API_TOKEN).campaigns.all()
+    """
+    client = get_mailchimp_client()
+    if not client:
+        return
```
Comment (pazembrz) [inline backend/inspirehep/mailing/providers/mailchimp.py]: Is it good idea to return silently here?

File: backend/inspirehep/mailing/providers/mailchimp.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        campaign_id (str): the campaign id to send.
+        html_content (str): the html content of the campaign.
+        test_emails (list): list of test emails.
+
+    Note:
+        If ``tests_emails`` is set the campaign will be send as test only.
+    """
+    client = get_mailchimp_client()
+    if not client:
+        return
```
Comment (pazembrz) [inline backend/inspirehep/mailing/providers/mailchimp.py]: also here?

File: backend/inspirehep/mailing/providers/mailchimp.py (hunk 3 above)
Comment (drjova) [reply backend/inspirehep/mailing/providers/mailchimp.py]: it's already throwing an exception https://github.com/inspirehep/inspirehep/pull/420/files#diff-a31ae48db9391f7531b5121831fc4833R21 but indeed this check is redundant

File: backend/inspirehep/mailing/api/jobs.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+from inspirehep.search.api import JobsSearch
+
+from ..providers.mailchimp import create_mailchimp_campaign, send_mailchimp_campaign
+
+
+def get_jobs_from_last_week():
+    """Jobs created the last 7 days."""
+    query = Q("range", **{"_created": {"gte": "now-7d/d", "lt": "now/d"}})
+    search = JobsSearch().query(query).sort("_created")
```
Comment (monaawi) [inline backend/inspirehep/mailing/api/jobs.py]: Just to make sure, is this the order we want the jobs to be sorted by? I know the task mentions ordered by date posted (and doesn't say ascending or descending).But in the example they seem to be descending.(so - _created)

File: backend/inspirehep/mailing/cli.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+
+@click.group()
+def mailing():
+    """Command to handle mailing."""
+
+
+@mailing.command(help="Sends a campaign with the INSPIRE jobs posted last week.")
+@click.option("-t", "--test", help="Send test campaign.", is_flag=True)
```
Comment (monaawi) [inline backend/inspirehep/mailing/cli.py]: would it be enough to have only --test-emails? I don't see the point of having both test and emails. If the user provides the emails, then we can automatically assume that he wants to send a test campaign

File: backend/inspirehep/mailing/cli.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+@click.option(
+    "-e",
+    "--email",
+    "emails",
+    help="Where to send the test email, ``--test`` is required.",
+    multiple=True,
+)
+@with_appcontext
+def send_weekly_jobs(test, emails):
+    if emails and not test:
```
Comment (pazembrz) [inline backend/inspirehep/mailing/cli.py]: And what will happen if --test will be set and --emails wont? User might thing that only test campaign will be sent. Why you require both parameters if you are not doing anything with test? Isn't better to just call `--email` parameter `--test-emails` and remove --test?

File: backend/inspirehep/mailing/providers/mailchimp.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+def send_mailchimp_campaign(campaign_id, html_content, test_emails=None):
+    """Send a given mailchimp campaign.
+
+    Args:
+        campaign_id (str): the campaign id to send.
+        html_content (str): the html content of the campaign.
+        test_emails (list): list of test emails.
+
+    Note:
+        If ``tests_emails`` is set the campaign will be send as test only.
```
Comment (monaawi) [inline backend/inspirehep/mailing/providers/mailchimp.py]: ```will be sent```

File: backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+                        cellspacing="0" cellpadding="0" border="0" align="left">
+                        <tbody>
+                            <tr>
+                                <td class="mcnTextContent"
+                                    style="padding-top:0; padding-right:18px; padding-bottom:9px; padding-left:18px;"
+                                    valign="top">
+                                    <a title="{{job.position}}"
+                                        href="https://labs.inspirehep.net/jobs/{{job.control_number}}"> {{job.position}}</a>
+                                    ({{ job.institutions[0].value}} - {{', '.join(job.regions)}}) [Deadline:
+                                    {{job.deadline_date}}] {{', '.join(job.ranks)}} - {{', '.join(job.arxiv_categories)}}
```
Comment (monaawi) [inline backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html]: In the example in the US, it displays also when the job was posted, do we not want that anymore?

File: backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+                    <table style="max-width:100%; min-width:100%;" class="mcnTextContentContainer" width="100%"
+                        cellspacing="0" cellpadding="0" border="0" align="left">
+                        <tbody>
+                            <tr>
+                                <td class="mcnTextContent"
+                                    style="padding-top:0; padding-right:18px; padding-bottom:9px; padding-left:18px;"
+                                    valign="top">
+                                    <a title="{{job.position}}"
+                                        href="https://labs.inspirehep.net/jobs/{{job.control_number}}"> {{job.position}}</a>
+                                    ({{ job.institutions[0].value}} - {{', '.join(job.regions)}}) [Deadline:
```
Comment (monaawi) [inline backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html]: institutions is not required by the schema, what happens if a job doesn't have institutions? Same for arxiv categories and ranks

File: backend/tests/integration/mailing/test_api_jobs.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+from inspirehep.mailing.api.jobs import (
+    get_jobs_from_last_week,
+    get_jobs_weekly_html_content,
+    send_jobs_weekly_campaign,
+)
+from inspirehep.records.api import InspireRecord
+
+
+def test_jobs_from_last_week(base_app, db, es_clear, cpreate_jobs):
```
Comment (monaawi) [inline backend/tests/integration/mailing/test_api_jobs.py]: ```create_jobs```

File: backend/tests/integration/mailing/test_cli.py [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def test_send_weekly_jobs_api_missing_exception(
+    app_cli_runner, base_app, db, es_clear, create_jobs
+):
+    with mock.patch.dict(base_app.config, {"MAILCHIMP_API_TOKEN": None}):
+        result = app_cli_runner.invoke(mailing, ["send_weekly_jobs"])
+        import ipdb
+
+        ipdb.set_trace()
```
Comment (monaawi) [inline backend/tests/integration/mailing/test_cli.py]: oops?

File: backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html (hunk 10 above)
Comment (drjova) [reply backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html]: are required in the submission, are schema is loose

File: backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html (hunk 9 above)
Comment (drjova) [reply backend/inspirehep/mailing/templates/mailing/jobs/mailchimp/job_list.html]: good catch totally missed it, I have updated the PR now

**Comment by drjova** [comment]
@pazembrz @monaawi updated

File: backend/scripts/server [hunk 13]
```diff
@@ -15,7 +15,7 @@ FLASK_ENV=development
 # Start Worker and Server. Whatchdog detects any change to the .py files and restarts celery
 poetry run watchmedo auto-restart -d . -p '*.py' -- celery worker -E -B -A inspirehep.celery -l INFO & pid_celery=$!
 
-poetry run gunicorn --reload -b 0.0.0.0:5000  --access-logfile "-" --error-logfile - inspirehep.wsgi:application & pid_server=$!
+poetry run gunicorn --reload -b 0.0.0.0:8000  --access-logfile "-" --error-logfile - inspirehep.wsgi:application & pid_server=$!
```
Comment (monaawi) [inline backend/scripts/server]: Why is this needed?

File: backend/tests/integration/mailing/conftest.py [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+from inspirehep.records.api import InspireRecord
+
+
+@pytest.fixture(scope="module")
+def app_config(app_config):
+    # Should be in this format format and length
+    # NOTE: Change this with the correct token to record new cassettes
+    app_config["MAILCHIMP_API_TOKEN"] = "5f5ee1097c0f1e705d2847dd14e8d6b2-us3"
+    # app_config["MAILCHIMP_API_TOKEN"] = "11111111111111111111111111111111-us3"
```
Comment (monaawi) [inline backend/tests/integration/mailing/conftest.py]: could you remove this comment?

File: backend/scripts/server (hunk 13 above)
Comment (drjova) [reply backend/scripts/server]: to match the ``SERVER_NAME`` otherwise the requests are not working
